### DISCUSSION TITLE
Want to lock down a private bucket

### DISCUSSION THREAD

**Discussion by csalmeida** [supabase/supabase#15097] [Questions] [answered]
What is the purpose of createSignedUploadUrl ?

I want a private bucket with RLS that restricts pretty much all access.

I don't have context for my user other than that they are authenticated and the last thing I want is anyone else listing, selecting or inserting in to the bucket.

The docs aren't clear but I assumed the signedUrl methods were to circumvent RLS, give me a way to upload without blanket upload access and same for downloading.

What are we missing, it seems there is really know difference between a private and public bucket.

**Comment by GaryAustin1** [comment]
None of the signedUrl calls to generate a signed URL work without meeting RLS.  They make a call to the server and check the policy.
Get publicURL is just a helper function and does not call the server at all, but then only works on a public bucket.

**Reply by tholder** [reply]
So, the obvious question is, what's the point of signing either an upload or download url if it still needs to respect RLS? Why not simply just use RLS? The docs aren't clear on the purpose of createSignedUploadUrl.

IMO, these need to skip RLS to give a single use (or rather TTL use) way to get a file in to a bucket. That way, my upload RLS policy for the bucket can be with check(false).

**Reply by GaryAustin1** [reply, marked as answer]
They are time limited.  The createSignedUploadUrl is only good for 2 hours.  The normal signedUrl calls you can set how long they are good for.  RLS only has to be met by the user generating the signedUrl.  Once that is done they can be used for what ever time period they are set to by anyone you give them to.

**Reply by tholder** [reply]
Ok. I think this is the critical piece of information that is missing form the docs. That the calls to the functions respect RLS but that the signedUrls then do not.

**Reply by inian** [reply]
This is documented in the client lib section @tholder . Here are the RLS policies that [createsigneduploadurl](https://supabase.com/docs/reference/javascript/storage-from-createsigneduploadurl) needs and that the resulting URL can be used without further authentication. [uploadtosignedurl](https://supabase.com/docs/reference/javascript/storage-from-uploadtosignedurl) mentions it too. 

Can the wording be improved or should this be mentioned in some other part of the docs?

**Comment by csalmeida** [comment]
Thanks @GaryAustin1
